### PR TITLE
create one commpg per pg

### PR DESCRIPTION
**Pull request by Sergei-Lebedev**

### CODE REVIEW

**Comment by kingchc** [comment]
resolved conflicts

File: src/torch_ucc.cpp [hunk 1]
```diff
@@ -596,18 +610,18 @@ ProcessGroupUCC::~ProcessGroupUCC() {
...
     	}
-    	if ((size_t)oob.store->add(oob.getKey("ucc_pg_closed"), 1) == eps.size()) {
-				std::vector<uint8_t> val = {1};
-      	oob.store->set(oob.getKey("ucc_pg_finished"), val);
-    	} else {
-      	oob.store->wait({oob.getKey("ucc_pg_finished")});
-    	}
+      if ((size_t)oob->store->add(oob->getKey("ucc_pg_closed"), 1) == eps.size()) {
+        std::vector<uint8_t> val = {1};
+        oob->store->add(oob->getKey("ucc_pg_finished"), val);
```
Comment (kingchc) [inline src/torch_ucc.cpp:621]: @Sergei-Lebedev - sorry I made a mistake when resolving the merge conflict and unable to push the fix here.
```suggestion
        oob->store->set(oob->getKey("ucc_pg_finished"), val);
```

Comment (Sergei-Lebedev) [reply src/torch_ucc.cpp]: fixed

File: src/torch_ucc.cpp [hunk 2]
```diff
@@ -622,21 +637,20 @@ ProcessGroupUCC::~ProcessGroupUCC() {
     comm->ucx_disconnect_eps(eps, &oob);
```
Comment (kingchc) [inline src/torch_ucc.cpp:637]: @Sergei-Lebedev - sorry I missed this when resolving merge conflict (again :( ). I've fixed it and tested the PR internally and everything looks good without perf. impact. Please help fix this compile issue, and I will merge this PR soon.

**Review by kingchc — APPROVED** [review]

**Pull request closed without merging**